backend/drm/drm.c

```diff
@@ -941,6 +993,8 @@ static void realloc_crtcs(struct wlr_drm_backend *drm, bool *changed_outputs) {
 		}
```

Comment:
(fuck github, for not letting one comment out of default visible context) --
one line above this is a return, so leaking things.

Personal opinion here but I don't really like the style of freeing exactly what's needed just as it's not needed anymore and am in the camp of freeing everything at the end with gotos jumping there on errors if possible. I guess either are fine, just got to be careful...

Reply:
Yes, immediate freeing has been a headache, especially in these busy functions. I've decided to free in blocks at the end of the function instead. I honestly don't see much advantage in immediately freeing resources anyway.